### PR TITLE
uprev to go 1.21

### PR DESCRIPTION
**Pull request by coconutruben**
This PR uprevs the `go.mod` to go 1.21

The motivation is fixing an issue with opentelemetry where newer
versions of the library will not compile unless the noop is added.
This change does the minimal edit required to get it to build again

Tested:
  go build ./...
  go test ./...

### CODE REVIEW

**Comment by marcoshuck** [comment]
Thanks Ruben for these changes!

We also need to:
- Change the go version in here: https://github.com/gazebo-web/gz-go/blob/main/.github/workflows/format.yaml
- And here (It should be set to `1.21` instead of `inputs.go-version`): https://github.com/gazebo-web/gz-go/blob/main/.github/workflows/test.yaml#L38

**Comment by coconutruben** [comment]
lmk if this addresses the issues

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version: ${{ inputs.go-version }}
+          go-version: 1.21
```
Comment (marcoshuck) [inline .github/workflows/test.yaml:38]:
```suggestion
          go-version: '1.21'
```

**Comment by marcoshuck** [comment]
> lmk if this addresses the issues

Pipelines are not running, something made the config invalid.

**Comment by marcoshuck** [comment]
> > lmk if this addresses the issues
> 
> Pipelines are not running, something made the config invalid since the first commit.

Ah my bad, this is a fork.

**Review by marcoshuck — APPROVED** [review]

**Pull request merged**